### PR TITLE
Mc 19 program trash can

### PR DESCRIPTION
**Pull request by haub94**
Add method to control the bin to method "MergeTwoComponents" and rename it to general function "HandleOverlappingObjects".

### CODE REVIEW

File: Merge Conflict/Assets/Scripts/ComponentHandler.cs [hunk 1]
```diff
@@ -81,29 +87,43 @@ private int GetHighestSpritePosition()
...
+        if (overlappedStaticObjects == null) { return; };
 
         //go trough all overlapped sprites and check if there is an mergabel one 
-        foreach (Collider2D staticComponent in overlappedStaticComponents)
+        foreach (Collider2D staticComponent in overlappedStaticObjects)
         {
-            //check if the overlapping sprite is an component too
-            if (staticComponent.gameObject != draggedComponent && IsComponent(staticComponent.gameObject))
+            //select the action according to the object-types (component, bin etc.)
+            if (staticComponent.gameObject != draggedComponent)
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/ComponentHandler.cs:105]: change to guard clause => if (x == y) { return; }

**Review by Xipit — CHANGES_REQUESTED** [review]
Looks good, only 1 small change requested

**Comment by Xipit** [comment]
Bug: currently checks for merging every frame when dragging, should only check when dragging ends

**Comment by haub94** [comment]
> Bug: currently checks for merging every frame when dragging, should only check when dragging ends

But for this case, it would not be possible/nice to implement the smooth "magic-over-another" animation. When we use the current code, the player moves the component to another/trashcan and at the near of the mergable component/trashcan the program overtakes the control and moves the last distance. 
On the other hand i understand your point, it would be more efficient to run the code after dragging. Your decision :)

**Pull request merged**